### PR TITLE
auto OCP-20932

### PR DESCRIPTION
**Pull request by yapei**
@yanpzhan  @xiaocwan  Please help review,thanks

### CODE REVIEW

File: lib/rules/web/admin_console/4.1/common_ui_elements.xyaml [hunk 1]
```diff
@@ -66,6 +66,10 @@ check_resource_name_and_icon:
   elements:
   - selector:
       xpath: //div[contains(@class,'co-m-pane__name')][/*[contains(.,'<icon_text>')] and /*[contains(., '<resource_name>')]]
+check_resource_link:
```
Comment (yapei) [inline lib/rules/web/admin_console/4.1/common_ui_elements.xyaml]: Used to check resource links defined in `a class=....href="/k8s/cluster/...."`

File: lib/rules/web/admin_console/4.1/common_ui_elements.xyaml [hunk 2]
```diff
@@ -211,6 +215,14 @@ check_resource_details_key_and_value:
   element: &key_value
     selector:
       xpath: //dt[contains(text(),'<key>')]/following-sibling::dd[1][contains(.,'<value>')]
+check_key_and_resource_link:
```
Comment (yapei) [inline lib/rules/web/admin_console/4.1/common_ui_elements.xyaml]: resourcenameandlink

Comment (yapei) [reply lib/rules/web/admin_console/4.1/common_ui_elements.xyaml]: Used to check resources like this

File: lib/rules/web/admin_console/4.1/common_ui_elements.xyaml [hunk 3]
```diff
@@ -211,6 +215,14 @@ check_resource_details_key_and_value:
...
+  element:
+    selector:
+      xpath: //dt[contains(text(),'<key>')]/following-sibling::dd[1]//a[contains(@href,'<resource_link>')]
+check_key_and_action_link:
+  element:
+    selector:
+      xpath: //dt[contains(text(),'<key>')]/following-sibling::dd[1]//button
```
Comment (yapei) [inline lib/rules/web/admin_console/4.1/common_ui_elements.xyaml]: tolerations
Used to check actions like this

File: lib/rules/web/admin_console/4.1/overview.xyaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  - selector:
+      xpath: //li//a[text()='Edit Environment']
+  - selector:
+      xpath: //li//a[text()='Edit Labels']
+  - selector:
+      xpath: //li//a[text()='Edit Annotations']
+  - selector:
+      xpath: //li//a[text()='Edit Deployment Config']
+  - selector:
+      xpath: //li//a[text()='Delete Deployment Config']
```
Comment (yanpzhan) [inline lib/rules/web/admin_console/4.1/overview.xyaml]: "check_available_action_menus" this rule is specially for dc, not adaptable to other resources, different resources have different action menus. We can change the rule name more precise for some resources, or update the rule to be a common one..

**Comment by xiaocwan** [comment]
LGTM. Thanks

File: lib/rules/web/admin_console/4.1/overview.xyaml (hunk 4 above)
Comment (yapei) [reply lib/rules/web/admin_console/4.1/overview.xyaml]: Agree with you, let me update it

Comment (yapei) [reply lib/rules/web/admin_console/4.1/overview.xyaml]: Updated to `check_dc_available_action_menus`

**Comment by yanpzhan** [comment]
@yapei  LGTM

**Comment by yapei** [comment]
@akostadinov  @pruan-rht  Can you help merge if no objections?

**Pull request merged**